### ISSUE TITLE
model updates and checks

### ISSUE THREAD

**Issue by kyle-messier** [Spatiotemporal-Exposures-and-Toxicology/NRT-AP-Model#422]
@mitchellmanware 
## Base learners
- [x] MLP 2 layer workflows are showing as untrained
- [x] Add dropout (0.1 - 0.3), particularly as we increase the hidden unit size
- [x] Fix penalty to 1e-6
- [x] Warning in `finetune` says that racing is being selected with `rmse` but we are using `mae` to select the best model. We should try to get racing to use `mae` 
- [ ] Are the epochs enough or too many? I can't figure out how to see the `mlp` training results by epoch. If we can see a few of them, we could get an idea of whether we are on point or need to increase/decrease

## Revisiting PCA 

- [ ] We need to add the PCA recipe to the output as per #420 at some point. As you said there, once we get the base and meta-learners to a good state.

## July 7, 2025
- [x] `cancel` -> `scancel`
- [x] Internal `parsnip::fit` in `fit_base_learner`
  - [ ]  Will need re-dispatch so "tuned" workflows can be applied to 
- [ ] @kyle-messier Explore targets unit tests with `targets::tar_assert_*`
- [ ] Run through `fit_meta_learner` to see performance with better performing base learners and State/Ecoregion dummy variables
- [ ] Calculate covariates at t-1 for covariate grid
- [ ] Calculate spatial neighbor values (TBD)

## Model updates to bring in-line with literature

- [ ] Run base learners into `parsnip::gen_additive_mod`--> Get single output (i.e. no ensemble or probalistic)
- [ ] Calculate all of the geographic covariates on regular spatial and temporal grid 
- [ ] Calculate **t-1** and **nearest-neighbor** "covariates" from the GAM prediction
- [ ] Run all of the base-learners with original + autocorrelated predictions 
- [ ] Decide on whether we do a probabilistic meta-learner (brms) or a monte carlo ensemble 

## CV sets

- [ ] Run parallel (w.r.t. pipeline, not cpus) base-learner/meta-learner pipeline with leave-one-location-out CV

**Comment by sigmafelix** [comment]
@kyle-messier We don't have any `future` dependency in our pipeline. In #424, I removed future and future.batchtools dependency from `DESCRIPTION`.

**Comment by mitchellmanware** [comment]
> Warning in finetune says that racing is being selected with rmse but we are using mae to select the best model. We should try to get racing to use mae

Patched here: https://github.com/NIEHS/beethoven/commit/8ec40017b9573185d2ab5d6db7951c5c64778ab5. Update reorders functions for user-defined metric and creates a function call for bare name, which is rquired by `yardstick::metric_set`.

S/o chatgpt for this one i was totally stumped

**Comment by mitchellmanware** [comment]
> Add dropout (0.1 - 0.3), particularly as we increase the hidden unit size
Fix penalty to 1e-6

https://github.com/NIEHS/beethoven/commit/5fb31ae208b8049b3924a6d8f2501d7f9123421e

**Comment by kyle-messier** [comment]
> > Add dropout (0.1 - 0.3), particularly as we increase the hidden unit size
> > Fix penalty to 1e-6
> 
> [5fb31ae](https://github.com/NIEHS/beethoven/commit/5fb31ae208b8049b3924a6d8f2501d7f9123421e)

@mitchellmanware  Typo - This needs to be `1e-6`, not `1e6` 😺
